### PR TITLE
feat: [DHIS2-11862] implement schedule info

### PR DESCRIPTION
**Pull request by jasminenguyennn**
### Schedule tabs
- Implement new `DataSection` as design
- Implement schedule info logic
- Calculate suggested schedule date

Screenshot 2021-10-11 at 12 59 38

### CODE REVIEW

File: src/core_modules/capture-core/components/WidgetEventSchedule/WidgetEventSchedule.container.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
+// @flow
+import React, { useMemo, useState } from 'react';
+import { useHistory } from 'react-router';
+import i18n from '@dhis2/d2-i18n';
+import { useDataMutation } from '@dhis2/app-runtime';
+import { getProgramAndStageForProgram, TrackerProgram } from '../../metaData';
+import { useOrganisationUnit } from '../WidgetEnrollmentEventNew/Validated/useOrganisationUnit';
```
[the remark this answers is not in the document]
Comment (JoakimSM) [reply src/core_modules/capture-core/components/WidgetEventSchedule/WidgetEventSchedule.container.js]: Thanks for implementing the changes here. Can we also add the fields data is retrieved to the state data here: https://github.com/dhis2/capture-app/blob/95042bf1f33099c34f1ba481c01f61e1fce9bf70/src/core_modules/capture-core/dataQueries/useOrganisationUnit.js#L40

File: src/core_modules/capture-core/components/WidgetEventSchedule/hooks/useDetermineSuggestedScheduleDate.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+            if (generatedByEnrollmentDate || !programConfig.displayIncidentDate) {
+                suggestedScheduleDate = moment(enrollmentDate).add(minDaysFromStart, 'days').format();
+            } else {
+                suggestedScheduleDate = moment(incidentDate).add(minDaysFromStart, 'days').format();
+            }
+            return suggestedScheduleDate;
+        },
+    ];
+    return scheduleDateComputeSteps.reduce((currentScheduleDate, computeScheduleDate) =>
+        (!currentScheduleDate ? computeScheduleDate() : convertClientToForm(currentScheduleDate, dataElementTypes.DATE))
```
Comment (JoakimSM) [inline src/core_modules/capture-core/components/WidgetEventSchedule/hooks/useDetermineSuggestedScheduleDate.js:82]: Please make sure scheduled date computed in the last step Is also converted

File: src/core_modules/capture-core/components/WidgetEventSchedule/hooks/useEventsInOrgUnit.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+            }),
+            [],
+        ),
+        { lazy: true },
+    );
+
+
+    useEffect(() => {
+        if (orgUnitId && selectedDate) {
+            refetch({ variables: { orgUnitId, selectedDate } });
```
Comment (JoakimSM) [inline src/core_modules/capture-core/components/WidgetEventSchedule/hooks/useEventsInOrgUnit.js:30]: It seems like you need to manually select a date for this one to trigger?

Comment (jasminenguyennn) [reply src/core_modules/capture-core/components/WidgetEventSchedule/hooks/useEventsInOrgUnit.js:30]: @JoakimSM I think it's sensible to wait for both orgUnit and selectedDate to be defined before doing the call? What do you suggest here?

Comment (JoakimSM) [reply src/core_modules/capture-core/components/WidgetEventSchedule/hooks/useEventsInOrgUnit.js:30]: If the date is auto-calculated (not manually selected/entered) we will also need to show the number of scheduled events. It didn't seem to do the request before I manually selected a date. (correct me if I'm wrong of course)

File: src/core_modules/capture-core/components/WidgetEventSchedule/hooks/useEventsInOrgUnit.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+        }
+    }, [orgUnitId, selectedDate]); // eslint-disable-line react-hooks/exhaustive-deps
```
Comment (simonadomnisoru) [inline src/core_modules/capture-core/components/WidgetEventSchedule/hooks/useEventsInOrgUnit.js:32]: Hey, 
Is there any reason why `refetch` is not in the dependency array?
Thanks!

**Review by simonadomnisoru — APPROVED** [review]
LGTM!

**Review by JoakimSM — APPROVED** [review]
Good work 👍

**Pull request merged**